extensions/iceberg/src/test/java/io/deephaven/iceberg/PyIcebergListColumnTest.java

```diff
@@ -0,0 +1,184 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg;
+
+import io.deephaven.engine.table.ColumnDefinition;
+import io.deephaven.engine.table.Table;
+import io.deephaven.engine.table.TableDefinition;
+import io.deephaven.engine.testutil.TstUtils;
+import io.deephaven.engine.util.TableTools;
+import io.deephaven.iceberg.sqlite.DbResource;
+import io.deephaven.iceberg.util.IcebergCatalogAdapter;
+import io.deephaven.iceberg.util.IcebergTableAdapter;
+import io.deephaven.qst.type.GenericType;
+import io.deephaven.qst.type.Type;
+import io.deephaven.vector.DoubleVector;
+import io.deephaven.vector.DoubleVectorDirect;
+import io.deephaven.vector.FloatVector;
+import io.deephaven.vector.FloatVectorDirect;
+import io.deephaven.vector.IntVector;
+import io.deephaven.vector.IntVectorDirect;
+import io.deephaven.vector.LongVector;
+import io.deephaven.vector.LongVectorDirect;
+import io.deephaven.vector.ObjectVector;
+import io.deephaven.vector.ObjectVectorDirect;
+import org.apache.iceberg.catalog.Namespace;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.nio.charset.StandardCharsets;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.net.URISyntaxException;
+import java.time.LocalTime;
+
+import static io.deephaven.util.QueryConstants.NULL_DOUBLE;
+import static io.deephaven.util.QueryConstants.NULL_FLOAT;
+import static io.deephaven.util.QueryConstants.NULL_INT;
+import static io.deephaven.util.QueryConstants.NULL_LONG;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * This test shows that Deephaven can read binary and list types written by pyiceberg. See TESTING.md and
+ * generate-pyiceberg-3.py for more details.
+ */
+@Tag("security-manager-allow")
+class PyIcebergListColumnTest {
+    private static final Namespace NAMESPACE = Namespace.of("list_test");
+    private static final TableIdentifier TABLE_IDENTIFIER = TableIdentifier.of(NAMESPACE, "data");
+
+    private static final TableDefinition TABLE_DEFINITION = TableDefinition.of(
```

Comment:
Let's call this `INFERRED_TD` since the source of truth is Iceberg in this case.